streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredWindowStore.java

```diff
@@ -88,7 +86,7 @@ public void init(final ProcessorContext context,
         // register and possibly restore the state from the logs
         final long startNs = time.nanoseconds();
         try {
-            inner.init(context, root);
+            super.init(context, root);
```

Comment:
Since super.init is `innerState.init(context, root);` could we just call that to be consistent with other calls, or are you intentionally call this in case we change the implementation of wrappedstore.init in the future?

Reply:
Good question...

Reply:
yes, it would be incorrect design to call `wrappedStore().init` directly. The job of `WrappedStateStore.init` is to "do the right thing". If we go and look at what it does today, and just do that thing, it'll work only until the superclass's implementation changes.

Another way to look at it is that we are "intercepting" the init call, to add some extra stuff before and after the init. This doesn't mean we get to interrupt the call hierarchy and circumvent our parent class, nor do we have any reason to do so.

Note that "the other calls" you refer to are not `StateStore` calls, they are `WindowStore` calls. And our superclass doesn't implement `WindowStore`, so it's not possible to call (eg) `super.put(key, value, timestamp)`.